plugins/catalog-react/src/hooks/useEntityListProvider.tsx

```diff
@@ -167,12 +173,20 @@ export const EntityListProvider = <EntityFilters extends DefaultEntityFilters>({
       }
 
       if (isMounted()) {
-        setSearchParams(
-          qs.stringify({ ...allQueryParams, filters: queryParams }),
-          {
-            replace: true,
-          },
+        const oldParams = qs.parse(window.location.search, {
+          ignoreQueryPrefix: true,
+        });
+        const newParams = qs.stringify(
+          { ...oldParams, filters: queryParams },
+          { addQueryPrefix: true },
         );
+        const newUrl = `${window.location.pathname}${newParams}`;
+        // We use direct history manipulation since useSearchParams and
+        // useNavigate in react-router-dom cause unnecessary extra rerenders.
+        // Also make sure to replace the state rather than pushing, since we
+        // don't want there to be back/forward slots for every single filter
+        // change.
```

Comment:
I do wonder if there's something we can do to solve this at the API level. It's not the first or last component that runs into this issue, and with this workaround I'm worried we might have an effect on analytics APIs and other things that might listen to react router navigation events. It's also bad that this happens in the first place as it ends up creating a bunch of entries in the page history as you've discovered.